src/pubsub/index.js

```diff
@@ -205,26 +221,28 @@ class PubsubBaseProtocol extends EventEmitter {
 
   /**
    * On an inbound stream opened.
-   * @private
+   *
+   * @protected
    * @param {Object} props
    * @param {string} props.protocol
-   * @param {DuplexIterableStream} props.stream
-   * @param {Connection} props.connection connection
+   * @param {MuxedStream} props.stream
+   * @param {Connection} props.connection - connection
    */
   _onIncomingStream ({ protocol, stream, connection }) {
     const peerId = connection.remotePeer
     const idB58Str = peerId.toB58String()
     const peer = this._addPeer(peerId, protocol)
     peer.attachInboundStream(stream)
 
-    this._processMessages(idB58Str, peer.inboundStream, peer)
+    peer.inboundStream && this._processMessages(idB58Str, peer.inboundStream, peer)
```

Comment:
It might be worth to make `attachInboundStream(stream)` return the value that it's assigning to the `this.inboundStream` that way this will turn into:

```js
const inboundStream = peer.attachInboundStream(stream)
this._processMessage(idB58Str, inboundStream, peer)
```

Reply:
yeah, this seems a good call